pkg/gofer/origins/openexchangerates.go

```diff
@@ -0,0 +1,96 @@
+//  Copyright (C) 2020 Maker Ecosystem Growth Holdings, INC.
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Affero General Public License as
+//  published by the Free Software Foundation, either version 3 of the
+//  License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Affero General Public License for more details.
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package origins
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/makerdao/oracle-suite/internal/query"
+)
+
+// Coinbase URL
+const openExchangeRatesURL = "https://openexchangerates.org/api/latest.json?app_id=%s&base=%s&symbols=%s"
+
+type openExchangeRatesResponse struct {
+	Timestamp intAsUnixTimestamp `json:"timestamp"`
+	Base      string             `json:"base"`
+	Rates     map[string]float64 `json:"rates"`
+}
+
+func NewOpenExchangeRates(pool query.WorkerPool, jsonParams json.RawMessage) *OpenExchangeRates {
```

Comment:
This package should not depend on any json stuff. The struct is clear and the caller should provide an already unmarshalled api key.

Reply:
Why not ? We don't know what parameters should be passed to concrete exchange. Might be there will be more stuff other than api key. Only exchange implementation knows what might be passed there. 
And in this case we have to parse json here or pass interface and typecast it somehow

Reply:
@konstantinzolotarev This is adding multiple responsibilities to the unit (i.e. package). When the package is used by other clients (and it is intended to be used by other projects even) it will require the caller to marshal a JSON to be able to use this constructor.  

I propose to create a separate `origins/config` or `origins/params` package to do the unmarshaling but still close to the struct being created.

Let's talk.